crypto/x509/x509_vfy.c

```diff
@@ -974,9 +994,11 @@ static int check_revocation(X509_STORE_CTX *ctx)
 {
     int i = 0, last = 0, ok = 0;
 
-    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK) == 0)
+    if (!(ctx->param->flags & X509_V_FLAG_CRL_CHECK)
+            && !(ctx->param->flags & X509_V_FLAG_OCSP_CHECK))
         return 1;
-    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK_ALL) != 0) {
+    if (ctx->param->flags & X509_V_FLAG_CRL_CHECK_ALL
+            || ctx->param->flags & X509_V_FLAG_OCSP_CHECK_ALL) {
```

Comment:
could be simplified : 
`if (ctx->param->flags & (X509_V_FLAG_CRL_CHECK_ALL|X509_V_FLAG_OCSP_CHECK_ALL) != 0) `